### PR TITLE
Finalizing documentation for `@itwin/imodels-client-management` and `@itwin/imodels-client-authoring`

### PR DESCRIPTION
**Pull request by austeja-bentley**
In this PR:
- Added missing documentation and unified the existing one in `@itwin/imodels-client-management` and `@itwin/imodels-client-authoring` packages.

### CODE REVIEW

**Review by robertas-kerpys — CHANGES_REQUESTED** [review]

File: clients/imodels-client-authoring/src/base/files/FileHandler.ts [hunk 1]
```diff
@@ -2,31 +2,86 @@
...
+
+/** Data reported on each file transfer iteration. */
 export interface ProgressData {
+  /** Bytes that are already transferred. */
   bytesTransferred: number;
+  /** Total size in bytes of the file being transferred. */
   bytesTotal: number;
 }
 
+/** Function that can be called to report file transfer progress. */
```
[the remark this answers is not in the document]
Comment (austeja-bentley) [reply clients/imodels-client-authoring/src/base/files/FileHandler.ts]: Fixed: https://github.com/iTwin/imodels-clients/blob/a58530989c58be7d08a19ab12e2369b643d03bc1/clients/imodels-client-authoring/src/base/files/FileHandler.ts#L14

File: clients/imodels-client-authoring/src/base/files/FileHandler.ts [hunk 2]
```diff
@@ -2,31 +2,86 @@
...
 export type ProgressCallback = (progressData: ProgressData) => void;
 
+/** Parameters for upload file operation. */
 export interface UploadFileParams {
+  /** Remote storage url where to upload the file. */
   uploadUrl: string;
+  /** Path of the local file to be uploaded. */
```
[the remark this answers is not in the document]
Comment (austeja-bentley) [reply clients/imodels-client-authoring/src/base/files/FileHandler.ts]: Fixed: https://github.com/iTwin/imodels-clients/blob/a58530989c58be7d08a19ab12e2369b643d03bc1/clients/imodels-client-authoring/src/base/files/FileHandler.ts#L21

File: clients/imodels-client-authoring/src/base/files/FileHandler.ts [hunk 3]
```diff
@@ -2,31 +2,86 @@
...
   sourceFilePath: string;
+  /** Function to be called to report progress on each transfer iteration. See {@link ProgressCallback}. */
   progressCallback?: ProgressCallback;
 }
 
 export interface DownloadFileParams {
```
[the remark this answers is not in the document]
Comment (austeja-bentley) [reply clients/imodels-client-authoring/src/base/files/FileHandler.ts:28]: Fixed: https://github.com/iTwin/imodels-clients/blob/a58530989c58be7d08a19ab12e2369b643d03bc1/clients/imodels-client-authoring/src/base/files/FileHandler.ts#L27

File: clients/imodels-client-authoring/src/base/files/FileHandler.ts [hunk 4]
```diff
@@ -2,31 +2,86 @@
...
+  /** Remote storage url from where to download the file. */
   downloadUrl: string;
+  /** Path of the local file to download to. */
```
[the remark this answers is not in the document]
Comment (austeja-bentley) [reply clients/imodels-client-authoring/src/base/files/FileHandler.ts]: Fixed: https://github.com/iTwin/imodels-clients/blob/a58530989c58be7d08a19ab12e2369b643d03bc1/clients/imodels-client-authoring/src/base/files/FileHandler.ts#L31

File: clients/imodels-client-authoring/src/operations/lock/LockOperationParams.ts [hunk 5]
```diff
@@ -5,16 +5,28 @@
 import { CollectionRequestParams, IModelScopedOperationParams } from "@itwin/imodels-client-management";
 import { LockedObjects } from "../../base";
 
+/** Url parameters supported in Lock list query. */
 export interface GetLockListUrlParams extends CollectionRequestParams {
+  /** Filters Locks for a specific Briefcase. */
```
[the remark this answers is not in the document]
Comment (austeja-bentley) [reply clients/imodels-client-authoring/src/operations/lock/LockOperationParams.ts]: Fixed: https://github.com/iTwin/imodels-clients/blob/a58530989c58be7d08a19ab12e2369b643d03bc1/clients/imodels-client-authoring/src/operations/lock/LockOperationParams.ts#L10

**Review by robertas-kerpys — APPROVED** [review]

**Pull request merged**